### PR TITLE
Create settings.yaml

### PR DESCRIPTION
**Pull request by alexanderankin**

### CODE REVIEW

File: .github/settings.yaml [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    # Branch Protection settings. Set to null to disable
+    protection:
+      # Required. Require at least one approving review on a pull request, before merging. Set to null to disable.
+      required_pull_request_reviews:
+        # The number of approvals required. (1-6)
+        required_approving_review_count: 1
+        # Dismiss approved reviews automatically when a new commit is pushed.
+        dismiss_stale_reviews: false
+        # Blocks merge until code owners have reviewed.
+        require_code_owner_reviews: false
```
Comment (balint-backmaker) [inline .github/settings.yaml:136]: I think we do need to make this `true` and add a `.github/CODEOWNERS` with this at the least:

```
.github/CODEOWNERS @testcontainers/oss-matinainers
.github/settings.yaml @testcontainers/oss-matinainers
```

Why we need this?
- the app that is running this sync process has an explicit security warning about this
- their suggested workaround is the above mentioned requirement plus the `CODEOWNERS` change

**Review by balint-backmaker — CHANGES_REQUESTED** [review]
one thing that I think we need to comply with - @kiview I think you'd agree but please lemme know

**Review by balint-backmaker — APPROVED** [review]
having seen your other message on slack, LGTM

**Review by totallyzen — APPROVED** [review]
having seen your other message on slack, LGTM

**Pull request merged**